Mathlib/RingTheory/MvPolynomial/Basic.lean

```diff
@@ -148,6 +149,21 @@ theorem linearIndependent_X : LinearIndependent R (X : σ → MvPolynomial σ R)
 set_option linter.uppercaseLean3 false in
 #align mv_polynomial.linear_independent_X MvPolynomial.linearIndependent_X
 
+instance {R σ : Type*} [CommSemiring R] [Finite σ] (N : ℕ) :
+    Module.Finite R (restrictDegree σ R N) :=
+  have := Finsupp.finite_setOf_bounded σ N
+  Module.Finite.of_basis (basisRestrictSupport R _)
+
+instance {R σ : Type*} [CommSemiring R] [Finite σ] (N : ℕ) :
+    Module.Finite R (restrictTotalDegree σ R N) :=
```

Comment:
(untested) I think it's good to keep R and σ explicit in case you need to manually do `haveI`.
```suggestion
instance [Finite σ] (N : ℕ) : Module.Finite R (restrictDegree σ R N) :=
  have := Finsupp.finite_setOf_bounded σ N
  Module.Finite.of_basis (basisRestrictSupport R _)

instance [Finite σ] (N : ℕ) : Module.Finite R (restrictTotalDegree σ R N) :=
```